cpp/src/neighbors/ivf_pq/ivf_pq_build.cuh

```diff
@@ -1632,9 +1660,24 @@ void extend(raft::resources const& handle,
   vec_batches.prefetch_next_batch();
   for (const auto& vec_batch : vec_batches) {
     const auto& idx_batch = *idx_batches++;
+    auto float_vec_batch =
+      raft::make_device_matrix<float, internal_extents_t>(handle, vec_batch.size(), index->dim());
+    raft::linalg::map_offset(
+      handle,
+      raft::make_device_vector_view<const T, internal_extents_t>(vec_batch.data(),
+                                                                 vec_batch.size() * index->dim()),
+      raft::make_device_vector_view<float, internal_extents_t>(float_vec_batch.data_handle(),
+                                                               vec_batch.size() * index->dim()),
+      [=] __device__(internal_extents_t idx, T i) { return utils::mapping<float>{}(i); });
```

Comment:
If you don't need offset as an argument, you can use just `linalg::map` in place of `linalg::map_offset`. In future, it would also be nice to fuse this with row_normalize in one op to avoid multiple passes through the data.